src/parser/parser.ts

```diff
@@ -2168,30 +2184,58 @@ function parseBinaryExpression(
       );
     }
 
-    if (parser.linterFlags & LinterFlags.NoBitwise) {
-      switch (parser.token) {
-        case SyntaxKind.BitwiseAnd:
-        case SyntaxKind.BitwiseAndAssign:
-        case SyntaxKind.BitwiseOr:
-        case SyntaxKind.BitwiseOrAssign:
-        case SyntaxKind.BitwiseXor:
-        case SyntaxKind.BitwiseXorAssign:
-        case SyntaxKind.ShiftLeft:
-        case SyntaxKind.ShiftLeftAssign:
-        case SyntaxKind.ShiftRight:
-        case SyntaxKind.ShiftRightAssign:
-        case SyntaxKind.LogicalShiftRight:
-        case SyntaxKind.LogicalShiftRightAssign:
+    if (parser.linterFlags & LinterFlags.No) {
+      if (parser.subRules && SubRules.DoubleEquals && left.kind !== SyntaxKind.NullKeyword) {
+        if (t === SyntaxKind.LooseEqual) {
+          parser.onError(
+            DiagnosticSource.Parser,
+            DiagnosticKind.Lint,
+            diagnosticMap[DiagnosticCode._may_be_unsafe_if_you_are_relying_on_type_coercion],
+            parser.curPos,
+            parser.pos
+          );
+          if (context & Context.Autofix) {
+            parser.token = SyntaxKind.StrictEqual;
+          }
```

Comment:
@aladdin-add This 'NoDoubleEquals' rule got an auto-fix. It looks correct to you? I added a test for it.

You are the ESLint expert here and I have no idea about the internals of ESLint except it's super slow and everything done wrong, so guide me if I do anything wrong while implementing this mess.